### PR TITLE
[11.x] Fix pluck() casting with qualified column

### PR DESCRIPTION
**Pull request by danharrin**
When using `pluck()` with an Eloquent builder, the results are cast / mutated.

However, when qualifying the column name with the table name (which is required when joining to tables that share a column name), the casting does not work, as Eloquent is not able to find casts with the qualified column name.

The solution is to strip out the table name from the qualified column name.

I have chosen to match the full `table.` string instead of just `.` in `after()`, since if the user did want to pluck using a duplicated column name on a joined table, we do not want to use the cast from the model.

### CODE REVIEW

**Comment by danharrin** [comment]
Fixed test `/Users/dan/Code/laravel-framework/tests/Integration/Database/EloquentBelongsToManyTest.php` which consumed this erroneous output from the `pluck()`, it didn't return the properly casted date.

File: tests/Integration/Database/EloquentBelongsToManyTest.php [hunk 1]
```diff
@@ -931,11 +931,7 @@ public function testCanTouchRelatedModels()
...
         foreach ($post->tags()->pluck('tags.updated_at') as $date) {
-            if ($this->driver === 'sqlsrv') {
-                $this->assertSame('2017-10-10 10:10:10.000', $date);
-            } else {
-                $this->assertSame('2017-10-10 10:10:10', $date);
-            }
+            $this->assertSame('2017-10-10 10:10:10', $date->toDateTimeString());
         }
 
         $this->assertNotSame('2017-10-10 10:10:10', Tag::find(2)->updated_at);
```
Comment (danharrin) [inline tests/Integration/Database/EloquentBelongsToManyTest.php:937]:
```suggestion
        $this->assertNotSame('2017-10-10 10:10:10', Tag::find(2)->updated_at?->toDateTimeString());
```

**Comment by taylorotwell** [comment]
@danharrin we may want to move this to `master` branch to be honest. Your test change shows how existing apps could be "expecting" this bug and already working around it.

**Comment by utsavsomaiya** [comment]
@danharrin, Is this PR affecting `contains` or `doesntContain` methods also?

image

**Comment by utsavsomaiya** [comment]
Okay!! On builder contains method is not available!!

Thanks!!

**Comment by taylorotwell** [comment]
@danharrin another weird edge case I'm not sure how we should handle. Imagine you have two models, both with the same accessor name, but the accessor behaves different on each model. Imagine like a "price" accessor which is slightly different between two models. If one model joined to the other and you selected the *other* model's "price", it would use the accessor from the model the query was initiated by, even though the value selected was the "price" from the other model.

**Comment by danharrin** [comment]
@taylorotwell I think this is handled when I matched the full `{$this->model->getTable()}.` instead of just `.`?

For `model.price`, `model.` will be stripped, so Eloquent looks for a `price` attribute, and it finds it.

For `model2.price`, `model.` will be stripped but it is not present in the column name, Eloquent looks for a `model2.price` attribute, and it cannot find it.

So in this case, the attribute on `model2` is not cast with the other model. It is not cast at all, which is the current behaviour when not using a qualified column name, I think?

**Comment by taylorotwell** [comment]
@danharrin got it - thanks

**Pull request merged**